pager/command.go

```diff
@@ -40,17 +49,20 @@ func (o Options) Run() error {
 		softWrap:            o.SoftWrap,
 		matchStyle:          o.MatchStyle.ToLipgloss(),
 		matchHighlightStyle: o.MatchHighlightStyle.ToLipgloss(),
-		timeout:             o.Timeout,
-		hasTimeout:          o.Timeout > 0,
-	}, tea.WithAltScreen()).Run()
+	}
+	_, err := tea.NewProgram(
+		m,
+		tea.WithAltScreen(),
+		tea.WithContext(ctx),
+	).Run()
 	if err != nil {
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			return exit.ErrTimeout
+		}
+		if errors.Is(err, tea.ErrInterrupted) {
+			return exit.ErrAborted
+		}
 		return fmt.Errorf("unable to start program: %w", err)
```

Comment:
Shouldn't this logic moved in the root command or main?

It's not logic you are copy pasting it everywhere the same dozen lines

But then it means the context should be created at top level and propagated, and not one created in each command

Reply:
yup, just did that :)